lib/vehicle_factory.rb

```diff
@@ -0,0 +1,67 @@
+class VehicleFactory
+  attr_reader :vehicles
+
+  def initialize
+    @vehicles = []
+  end
+
+  def create_vehicles(registrations)
+    registrations.map do |registration|
+      new_vehicle = Vehicle.new({
+      vin: registration[:vin_1_10],
+      year: registration[:model_year],
+      make: registration[:make],
+      model: registration[:model],
+      engine: :ev, county: registration[:county]})
+      vehicles << new_vehicle
+    end
+  end
+  
+  def create_boats(registrations)
+    registrations.map do |registration|
+      new_vehicle = Vehicle.new({
+      vin: registration[:vin],
+      year: registration[:model_year],
+      make: registration[:make],
+      model: "unknown",
+      engine: :BOAT, county: registration[:county]})
+      vehicles << new_vehicle
+    end
+  end
```

Comment:
I mentioned this earlier but it's worth mentioning again, `map` does not require a shoveling at the end.